skimage/morphology/footprints.py

```diff
@@ -95,6 +99,126 @@ def footprint_from_sequence(footprints):
     return morphology.binary_dilation(imag, footprints)
 
 
+class FootprintComponent(tuple):
```

Comment:
Why do we need a whole class for this?

Reply:
Not necessarily. I'm still feeling out what feels right to me here. Nevertheless, using a custom immutable container can some nice advantages over a function. E.g. it's a good place to document that is also picked up by many inspection tools like IDEs and when using the data interactively. 

And as mentioned in the PR summary, I might pull this out into a separate PR if this get's to be its own thing.

Reply:
Just to explain my thinking: when I read this class, it's immediately a lot to take in: why is this thing here, what is it doing, etc. It is a burden for maintainers to comprehend code concepts, and what they do. If you have widely used patterns across a library, it gets easier.

I'd say: use the structures you get the functionality you want, but no more. And if you have to sacrifice a structure and some small convenience, to make the code a lot shorter and easier to parse, do that as well. Consider alternatives, e.g. namedtuple is lighter than data class, etc. But simplicity should be at the forefront of our considerations, even if we choose not to embrace it for any individual instance.

Reply:
I can follow your reasoning but probably not your weighting of pros and cons with regard to this. I'm all for simplicity. But I'd disagree that using simple containers only is always guaranteed to lead to simple to understand code. E.g. in my opinion a deeply nested or long container with homogeneous objects quickly becomes a comprehension burden for maintainers itself. 

That said, `tuple[tuple[ndarray, int], ...]` isn't too complicated a data structure. And in [`_footprint_is_sequence`](https://github.com/scikit-image/scikit-image/blob/8e37f77568b580be56424adc3ac4b5515da3ab2e/skimage/morphology/footprints.py#L25), we've already builtin validation. So, I'll scratch this from the current PR. :)

Reply:
Happy to revisit this at a later point; in the end, it's an aesthetic preference.